service/src/main/java/tds/config/services/ConfigService.java

```diff
@@ -0,0 +1,38 @@
+package tds.config.services;
+
+import tds.config.ClientSystemFlag;
+import tds.config.ClientTestProperty;
+
+import java.util.NoSuchElementException;
+
+/**
+ * Provide an interface for interacting with the {#code configs} database and providing configuration information.
+ */
+public interface ConfigService {
+    /**
+     * Get all the {@link ClientTestProperty} records from the {@code configs.client_testproperties} table for the
+     * client and test.
+     *
+     * @param clientName The name of the client (typically SBAC or SBAC_PT)
+     * @param assessmentId The assessment identifier
+     * @return A {@code ClientTestProperty} for the specified client name and assessment id.
+     * @throws NoSuchElementException if there is no {@code ClientTestProperty} record for the specified client name and
+     * assessment id.
+     */
+    ClientTestProperty getClientTestProperty(String clientName, String assessmentId) throws NoSuchElementException;
+
+    /**
+     * Get the {@link ClientSystemFlag} for the specified client and audit name.
+     * <p>
+     *     The audit name is the name/description of the ClientSystemFlag.  Ideally the source of data for this method
+     *     will be an in-memory collection; the {@code ClientSystemFlag} values rarely change, making them a strong
+     *     candidate for caching.
+     * </p>
+     *
+     * @param clientName  The name of the client (typically SBAC or SBAC_PT)
+     * @param auditObject The name of the desired {@code ClientSystemFlag}
+     * @return The {@code ClientSystemFlag} that matches the client and audit name.
+     * @throws NoSuchElementException if the {@code ClientSystemFlag} with the specified audit name does not exist
+     */
+    ClientSystemFlag getClientSystemFlag(String clientName, String auditObject) throws NoSuchElementException;
```

Comment:
We discussed a bit offline.  Should be throwing Exceptions when we try to retrieve something from a service if it isn't available?  It then makes any caller surround their service call with a try/catch and handle the exception.  Especially for the case where if the ClientSystemFlag isn't available look for a property value for the flag as an example.  

What if instead we returned an [Optional](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html) and allow the caller to throw an Exception if that is the intended consequence.